pinot-core/src/main/java/org/apache/pinot/core/common/datatable/BaseDataTable.java

```diff
@@ -0,0 +1,300 @@
+    int[] ints = new int[length];
+    for (int i = 0; i < length; i++) {
+      ints[i] = _variableSizeData.getInt();
+    }
+    return ints;
+  }
+
+  @Override
+  public long[] getLongArray(int rowId, int colId) {
+    int length = positionCursorInVariableBuffer(rowId, colId);
+    long[] longs = new long[length];
+    for (int i = 0; i < length; i++) {
+      longs[i] = _variableSizeData.getLong();
+    }
+    return longs;
+  }
+
+  @Override
+  public float[] getFloatArray(int rowId, int colId) {
+    int length = positionCursorInVariableBuffer(rowId, colId);
+    float[] floats = new float[length];
+    for (int i = 0; i < length; i++) {
+      floats[i] = _variableSizeData.getFloat();
+    }
+    return floats;
+  }
+
+  @Override
+  public double[] getDoubleArray(int rowId, int colId) {
+    int length = positionCursorInVariableBuffer(rowId, colId);
+    double[] doubles = new double[length];
+    for (int i = 0; i < length; i++) {
+      doubles[i] = _variableSizeData.getDouble();
+    }
+    return doubles;
+  }
+
+  @Override
+  public String[] getStringArray(int rowId, int colId) {
+    int length = positionCursorInVariableBuffer(rowId, colId);
+    String[] strings = new String[length];
+    Map<Integer, String> dictionary = _dictionaryMap.get(_dataSchema.getColumnName(colId));
+    for (int i = 0; i < length; i++) {
+      strings[i] = dictionary.get(_variableSizeData.getInt());
+    }
+    return strings;
+  }
+
+  private int positionCursorInVariableBuffer(int rowId, int colId) {
+    _fixedSizeData.position(rowId * _rowSizeInBytes + _columnOffsets[colId]);
+    _variableSizeData.position(_fixedSizeData.getInt());
+    return _fixedSizeData.getInt();
+  }
+
+  @Override
+  public String toString() {
+    if (_dataSchema == null) {
+      return _metadata.toString();
+    }
+
+    StringBuilder stringBuilder = new StringBuilder();
+    stringBuilder.append(_dataSchema.toString()).append('\n');
+    stringBuilder.append("numRows: ").append(_numRows).append('\n');
+
+    ColumnDataType[] storedColumnDataTypes = _dataSchema.getStoredColumnDataTypes();
+    _fixedSizeData.position(0);
+    for (int rowId = 0; rowId < _numRows; rowId++) {
+      for (int colId = 0; colId < _numColumns; colId++) {
+        switch (storedColumnDataTypes[colId]) {
+          case INT:
+            stringBuilder.append(_fixedSizeData.getInt());
+            break;
+          case LONG:
+            stringBuilder.append(_fixedSizeData.getLong());
+            break;
+          case FLOAT:
+            stringBuilder.append(_fixedSizeData.getFloat());
+            break;
+          case DOUBLE:
+            stringBuilder.append(_fixedSizeData.getDouble());
+            break;
+          case STRING:
+            stringBuilder.append(_fixedSizeData.getInt());
+            break;
+          // Object and array.
+          default:
+            stringBuilder.append(String.format("(%s:%s)", _fixedSizeData.getInt(), _fixedSizeData.getInt()));
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

This code uses &#39;%s&#39; to format int: &#39;getInt&#39; expression. This is a potential locale-sensitive handling issue. It might cause errors in the handling and processing of the statement at line: 291. Consider formatting this data with &#39;%d&#39; instead.